### PR TITLE
Rework the mangareader multisrc

### PR DESCRIPTION
**Pull request by Secozzi**
A lot of websites using the mangareader theme has popped up since the multisrc theme was made, so this rework aims to make it easier to add them.

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: multisrc/overrides/mangareader/comickiba/src/Manhuagold.kt [hunk 1]
```diff
@@ -2,33 +2,19 @@ package eu.kanade.tachiyomi.extension.en.comickiba
...
-    override val name = "Manhuagold"
-
-    override val lang = "en"
-
-    override val baseUrl = "https://manhuagold.com"
+class Manhuagold : MangaReader(
+    "Manhuagold",
+    "https://manhuagold.top",
+    "en",
+    "views",
```
Comment (AwkwardPeak7) [inline multisrc/overrides/mangareader/comickiba/src/Manhuagold.kt]: explicitly specify `views` paramter for better readibility

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt [hunk 2]
```diff
@@ -62,66 +178,271 @@ abstract class MangaReader : HttpSource(), ConfigurableSource {
...
+     * Method to create request for either chapters or volumes
+     */
+    open fun chapterListRequest(mangaUrl: String, type: String): Request =
+        throw NotImplementedError()
+
+    /**
+     * Method to parse elements depending on type
+     */
+    open fun parseChapterElements(response: Response, isVolume: Boolean): List<Element> =
+        throw NotImplementedError()
```
Comment (AwkwardPeak7) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt]: if sources are expected to implement these methods, then make them abstract

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: multisrc/overrides/mangareader/mangafire/src/MangaFire.kt [hunk 3]
```diff
@@ -26,109 +25,85 @@ import java.util.Locale
...
-) : MangaReader() {
-    override val name = "MangaFire"
-
-    override val baseUrl = "https://mangafire.to"
+) : MangaReader(
+    "MangaFire",
+    "https://mangafire.to",
+    lang,
+    "most_viewed",
+    "recently_updated",
```
Comment (AwkwardPeak7) [inline multisrc/overrides/mangareader/mangafire/src/MangaFire.kt]: here too

**Review by stevenyomi — CHANGES_REQUESTED** [review]
Move `open val`s to the constructor like #931.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt [hunk 4]
```diff
@@ -1,35 +1,157 @@
...
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
-import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
```
Comment (stevenyomi) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt]: This source is not a conventional parsed HTTP source. Do not inherit `ParsedHttpSource` just to reuse the "field/functions names". Declare your own ones instead.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt [hunk 5]
```diff
@@ -62,66 +178,268 @@ abstract class MangaReader : HttpSource(), ConfigurableSource {
...
+    open fun String?.getStatus(): Int = when (this?.lowercase()) {
+        "ongoing" -> SManga.ONGOING
+        "publishing" -> SManga.ONGOING
+        "releasing" -> SManga.ONGOING
+        "completed" -> SManga.COMPLETED
+        "finished" -> SManga.COMPLETED
+        "on-hold" -> SManga.ON_HIATUS
+        "on_hiatus" -> SManga.ON_HIATUS
+        "discontinued" -> SManga.CANCELLED
+        "canceled" -> SManga.CANCELLED
```
Comment (stevenyomi) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangareader/MangaReader.kt]: Use comma to join conditions with same result.

File: multisrc/overrides/mangareader/mangafire/src/MangaFire.kt [hunk 6]
```diff
@@ -26,109 +25,85 @@ import java.util.Locale
...
+    override fun searchMangaSelector() = ".original.card-lg .unit .inner"
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        element.selectFirst(".info > a")!!.let {
+            setUrlWithoutDomain(it.attr("abs:href"))
+            title = it.ownText()
+        }
+        element.selectFirst("img")!!.let {
+            thumbnail_url = it.imgAttr()
         }
```
Comment (stevenyomi) [inline multisrc/overrides/mangareader/mangafire/src/MangaFire.kt]:
```suggestion
        thumbnail_url = element.selectFirst("img")!!.imgAttr()
```

**Review by stevenyomi — CHANGES_REQUESTED** [review]

**Comment by stevenyomi** [comment]
Let's merge this after the migration. It's too painful to edit in the current project architecture.

**Comment by stevenyomi** [comment]
Continued in #1241

**Pull request closed without merging**